smoke-tests-otel-starter/src/test/java/io/opentelemetry/smoketest/OtelSpringStarterSmokeTest.java

```diff
@@ -0,0 +1,125 @@
+
+@SpringBootTest(
+    classes = {
+      OtelSpringStarterSmokeTestApplication.class,
+      OtelSpringStarterSmokeTest.TestConfiguration.class
+    },
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    properties = {"otel.exporter.otlp.enabled=false", "otel.metric.export.interval=1"
+      // 1 minute by default
+    })
+class OtelSpringStarterSmokeTest {
+
+  public static final InMemoryMetricExporter METRIC_EXPORTER =
+      InMemoryMetricExporter.create(AggregationTemporality.DELTA);
+  private static final InMemoryLogRecordExporter LOG_RECORD_EXPORTER =
+      InMemoryLogRecordExporter.create();
+  public static final InMemorySpanExporter SPAN_EXPORTER = InMemorySpanExporter.create();
+
+  @Autowired private TestRestTemplate testRestTemplate;
+
+  @Configuration(proxyBeanMethods = false)
+  static class TestConfiguration {
+    @Bean
+    public MetricExporter metricExporter() {
+      return METRIC_EXPORTER;
+    }
+
+    @Bean
+    public SpanExporter spanExporter() {
+      return SPAN_EXPORTER;
+    }
+
+    @Bean
+    public LogRecordExporter logRecordExporter() {
+      return LOG_RECORD_EXPORTER;
+    }
+  }
+
+  @Test
+  void shouldSendTelemetry() throws InterruptedException {
+
+    testRestTemplate.getForObject(OtelSpringStarterSmokeTestController.URL, String.class);
+
+    Thread.sleep(5_000); // Sleep time could be potentially reduced and perhaps removed with
+    // https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/8962
+    // and https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/8963
+
+    List<SpanData> exportedSpans = SPAN_EXPORTER.getFinishedSpanItems();
+
+    // Span
+    TracesAssert.assertThat(exportedSpans)
+        .hasSize(2)
+        .hasTracesSatisfyingExactly(
+            traceAssert ->
+                traceAssert.hasSpansSatisfyingExactly(
+                    spanDataAssert ->
+                        spanDataAssert
+                            .hasKind(SpanKind.CLIENT)
+                            .hasAttribute(
+                                SemanticAttributes.DB_STATEMENT,
+                                "create table test_table (id bigint not null, primary key (id))")),
+            traceAssert ->
+                traceAssert.hasSpansSatisfyingExactly(
+                    spanDataAssert ->
+                        spanDataAssert
+                            .hasKind(SpanKind.SERVER)
+                            .hasAttribute(SemanticAttributes.HTTP_METHOD, "GET")
+                            .hasAttribute(SemanticAttributes.HTTP_STATUS_CODE, 200L)
+                            .hasAttribute(SemanticAttributes.HTTP_ROUTE, "/ping")));
+
+    // Metric
+    List<MetricData> exportedMetrics = METRIC_EXPORTER.getFinishedMetricItems();
+    assertThat(containsTestHistogramMetric(exportedMetrics))
+        .as("Should contain " + OtelSpringStarterSmokeTestController.TEST_HISTOGRAM + " metric.")
+        .isTrue();
```

Comment:
Can you use `.anySatisfy()` instead? It should result in a better assertion message in case it fails